pkg/arvo/sys/vane/ames.hoon

```diff
@@ -2136,10 +2137,27 @@
     ::
     (emit unix-duct.ames-state %give %turf turfs)
   ::  +on-vega: handle kernel reload
-  ::  +on-trim: handle request to free memory
   ::
   ++  on-vega  event-core
-  ++  on-trim  event-core
+  ::  +on-trim: handle request to free memory
+  ::
+  ::  %ruin comets not seen for six months
+  ::
+  ++  on-trim
+    ^+  event-core
+    =;  rui=(set @p)
+      (emit duct %pass /ruin %j %ruin rui)
+    =-  (silt (turn - head))
+    %+  skim
+      ~(tap by peers.ames-state)
+    |=  [=ship s=ship-state]
+    ?.  &(?=(%known -.s) =(%pawn (clan:title ship)))  %.n
+    ?&  (gth (sub now ~d180) last-contact.qos.s)
+        ::
+        %+  levy  (turn ~(tap by snd.s) tail)
+        |=  m=message-pump-state
+        ?~(unsent-fragments.m %.y %.n)
```

Comment:
I believe this condition is reversed, right?  We want to check that unsent-fragments is not null for at least one bone.

I would also tend to use +any:by and ignore the first argument rather than use turn+tap+levy, but that's a stylistic thing.

```suggestion
        %-  ~(any by snd.s)
        |=  [* m=message-pump-state]
        !=(~ unsent-fragments.m)
```

Reply:
Looks like `+any:by` only iterates over the values, so it's even better here. Fixed.